fuzz/build.rs

```diff
@@ -0,0 +1,148 @@
+fn main() -> anyhow::Result<()> {
+    component::generate_static_api_tests()?;
+
+    Ok(())
+}
+
+mod component {
+    use anyhow::{anyhow, Context, Error, Result};
+    use arbitrary::{Arbitrary, Unstructured};
+    use component_fuzz_util::{self, Declarations, TestCase};
+    use proc_macro2::TokenStream;
+    use quote::{format_ident, quote};
+    use rand::rngs::StdRng;
+    use rand::{Rng, SeedableRng};
+    use std::env;
+    use std::fmt::Write;
+    use std::fs;
+    use std::iter;
+    use std::path::PathBuf;
+    use std::process::Command;
+
+    pub fn generate_static_api_tests() -> Result<()> {
+        println!("cargo:rerun-if-changed=build.rs");
+        let out_dir = PathBuf::from(
+            env::var_os("OUT_DIR").expect("The OUT_DIR environment variable must be set"),
+        );
+
+        let mut out = String::new();
+        write_static_api_tests(&mut out)?;
+
+        let output = out_dir.join("static_component_api.rs");
+        fs::write(&output, out)?;
+
+        drop(Command::new("rustfmt").arg(&output).status());
+
+        Ok(())
+    }
+
+    fn write_static_api_tests(out: &mut String) -> Result<()> {
+        let seed = if let Ok(seed) = env::var("WASMTIME_FUZZ_SEED") {
+            seed.parse::<u64>()
+                .with_context(|| anyhow!("expected u64 in WASMTIME_FUZZ_SEED"))?
+        } else {
+            StdRng::from_entropy().gen()
+        };
```

Comment:
Could the seed be printed here to assist debugging in case the build script later returns an error?